pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -82,6 +84,18 @@ func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.Rele
 // Name returns Name for the precondition.
 func (pf *Upgradeable) Name() string { return "ClusterVersionUpgradeable" }
 
+// getCurrentVersion determines and returns the cluster's current version by iterating through the
+// provided update history until it finds the first version with update State of Completed.
+func getCurrentVersion(history []configv1.UpdateHistory) string {
+	for _, h := range history {
+		if h.State == configv1.CompletedUpdate {
+			klog.V(5).Infof("Cluster current version=%s", h.Version)
+			return h.Version
+		}
+	}
+	return ""
```

Comment:
In the case where we've never had a completed update, we probably want to use the tail history entry, which is the install-time version, instead of an empty string.  We know there will be no older in-cluster components, and it will allow folks to retarget within a z-stream to unblock a hung install (not that that seems particularly likely, but still, no reason we should block it in preconditions).

Reply:
Changed in [9a7b056](https://github.com/openshift/cluster-version-operator/pull/394/commits/9a7b056e8003c358b4eb72e174dcccfcdbdb9196)